### PR TITLE
Implement location cookies

### PR DESCRIPTION
**Pull request by nssafir**
Use Navigator API to create location cookies if user chooses to share location and implement front-end to ask user. The cookies do not affect the request to the civic servlet yet (that will implemented in the next PR).

### CODE REVIEW

File: src/main/webapp/index.html [hunk 1]
```diff
@@ -16,6 +17,9 @@
       <div id="form-title">
         <p>Copy your Tweet here</p>
       </div>
+      <label><input type="checkbox" onclick="setLocationCookie()" /> Use Location</label>
```
Comment (juanuribe28) [inline src/main/webapp/index.html]: [Nit] Input tag should be outside the label.

File: src/main/webapp/index.html [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <link rel="stylesheet" href="/styles/base.css">
     <link rel="stylesheet" href="/styles/index.css">
     <link href="https://fonts.googleapis.com/css2?family=Open+Sans&family=Oswald&display=swap" rel="stylesheet">
+    <script src="scripts/cookies.js"></script>
```
Comment (juanuribe28) [inline src/main/webapp/index.html]: [Optional] Move the script tag to the end of the body, as it isn't needed during loading.

File: src/main/webapp/index.html (hunk 1 above)
Comment (juanuribe28) [inline src/main/webapp/index.html]: [Optional] Is best practice to keep javascript separate from markup, so I would remove the onclick attribute and instead add a event listener to javascript, or even better use jQuery.

File: src/main/webapp/scripts/cookies.js [hunk 3]
```diff
@@ -82,3 +82,19 @@ function loadCheckboxStateCookie(cookieName) {
...
+ */
+function setLocationCookie() {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+        setCookie("latitude", position.coords.latitude, 1);
+        setCookie("longitude", position.coords.longitude, 1);
+      }, function(err) {
+        console.warn(`ERROR(${err.code}): ${err.message}`);
+        document.getElementById("location-warning")
+            .innerHTML = "We cannot access your location. Try checking your browswer settings";
```
Comment (juanuribe28) [inline src/main/webapp/scripts/cookies.js]: [Optional] Instead of changing the DOM I would send an alert to the user.

**Review by juanuribe28 — CHANGES_REQUESTED** [review]

File: src/main/webapp/index.html (hunk 1 above)
Comment (nssafir) [reply src/main/webapp/index.html]: Which javascript file would this be in?

Comment (juanuribe28) [reply src/main/webapp/index.html]: index.js. If it doesn't exist, create it and add the function.

Comment (nssafir) [reply src/main/webapp/index.html]: This is the only way I am able to get the label and checkbox to appear inline with each other on the website. Let me know if you know another way.

Comment (juanuribe28) [reply src/main/webapp/index.html]: You can use the for attribute https://www.w3schools.com/tags/tag_label.asp

Comment (nssafir) [reply src/main/webapp/index.html]: That's still not working

Comment (juanuribe28) [reply src/main/webapp/index.html]: That's weird

File: src/main/webapp/styles/index.js [hunk 4]
```diff
@@ -0,0 +1 @@
+$("checkbox").click(setLocationCookie());
```
Comment (juanuribe28) [inline src/main/webapp/styles/index.js]: [Nit] remove callback function parentheses and add new line.

**Review by juanuribe28 — CHANGES_REQUESTED** [review]

**Review by juanuribe28 — APPROVED** [review]

**Review by daperlmut — APPROVED** [review]

**Pull request merged**